### PR TITLE
Convert `get-page-targeting` to TypeScript

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Convert `get-page-targeting.js` to TypeScript, now that the `messenger` is typed (see #24553).

Github unfortunately struggles to track the file rename as the percentage content is changed so much in one PR. The diff from JS to TS for `get-page-targeting.js` is shown below.

```diff
diff --git a/static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.js b/static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts
index c98f519c8e..80d5ca4ee9 100644
--- a/static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.js
+++ b/static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts
@@ -1,7 +1,15 @@
-import config from '../../../../lib/config';
+import type { RegisterListener } from '../messenger';
 
-const init = (register) => {
-	register('get-page-targeting', () => config.get('page.sharedAdTargeting'));
+/**
+ * Register a listener for iframes to request shared ad targeting
+ *
+ * Allows for ads to be served into SafeFrame whilst retaining the ability to define a passback
+ */
+const init = (register: RegisterListener): void => {
+	register(
+		'get-page-targeting',
+		() => window.guardian.config.page.sharedAdTargeting,
+	);
 };
 
 export { init };
```

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

More types!

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { RegisterListener } from '../messenger';
+
+/**
+ * Register a listener for iframes to request shared ad targeting
+ *
+ * Allows for ads to be served into SafeFrame whilst retaining the ability to define a passback
+ */
+const init = (register: RegisterListener): void => {
+	register(
+		'get-page-targeting',
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts:10]: Could we narrow the allowed string types here to a known list? Could be another PR.

**Review by arelra — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts (hunk 1 above)
Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts:10]: Do you the mean the strings you can register for, [like this](https://github.com/guardian/frontend/blob/8dbaec95d4fec050dc5f5023f829c98b847c26fb/static/src/javascripts/projects/commercial/modules/messenger.ts#L7-L19)?

Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/messenger/get-page-targeting.ts:10]: Ah yes, so it's already covered by the previous conversion! Thanks
